jwtmiddleware.go

```diff
@@ -23,10 +25,10 @@ type TokenExtractor func(r *http.Request) (string, error)
 
 // Options is a struct for specifying configuration options for the middleware.
 type Options struct {
-	// The function that will return the Key to validate the JWT.
-	// It can be either a shared secret or a public key.
-	// Default value: nil
-	ValidationKeyGetter jwt.Keyfunc
+	// TODO(jayhelton) add comment
+	ValidationOptions []jwt.ValidateOption
```

Comment:
Doubting if we should put abstraction in place instead of leaking the underlying JWT implementation used.  If we add an abstraction, we can more easily switch JWT library used in future.

Reply:
I think a healthy abstraction is a better idea! I'll add a commit for the suggestion this week. Open to any and all suggestions! 😁

Reply:
I started looking at making a deeper abstraction, but started out with less of an abstraction and more of removal of claim validation responsibilities from the middleware. In this iteration, seen in the below commit, i added a ValidatorFunc as a replacement for the ValidationOptions array, where the server can implement its own jwt validation rules, either with jwx or not. 

https://github.com/auth0/go-jwt-middleware/pull/76/commits/f0ece3c6e3ac16c967798dfc46c4f84a8f8bd94c

Reply:
If we liked it, we could keep the `ValidationKeyGetter`, where its a func that performs the validation operations and just returns the key.

I had found the `ValidationKeyGetter` to be a little awkward, because it was a function that had two responsibilities, validating claims, and returning the key for signature verification. I wanted to try a couple things while we had the chance 🚀